src/OrchardCore.Modules/OrchardCore.Autoroute/Services/AutorouteShapeTableProvider.cs

```diff
@@ -11,25 +12,29 @@ public class AutorouteShapeTableProvider : ShapeTableProvider
     public override ValueTask DiscoverAsync(ShapeTableBuilder builder)
     {
         builder.Describe("Content")
-            .OnDisplaying(displaying =>
-            {
-                var shape = displaying.Shape;
-                var contentItem = shape.GetProperty<ContentItem>("ContentItem");
+            .OnDisplaying(displaying => AddAlternates(displaying, "Content"));
+        builder.Describe("Widget")
+            .OnDisplaying(displaying => AddAlternates(displaying, "Widget"));
```

Comment:
I don't think this should be added here. `Widget` is part of the Widgets module. I think the Widget logic should be added to a new ShapeTableProvider in the Widgets module and registered like this

```
[RequireFeatures("OrchardCore.Autoroute")]
public sealed AutorouteStartup : StartupBase
{
    public override void ConfigureServices(IServiceCollection services)
    {
        services.AddScoped<IShapeTableProvider, AutorouteShapeTableProvider>()
    }
}
```

so it would be part of the widgets module but only available if the Autoroute feature is also enabled.

Reply:
this would require setting a reference to autoroute and alias module from the widgets module because of var aliasPart = contentItem?.As<AliasPart>();.
How about leave it here  in separate feature but use : [RequireFeatures("OrchardCore.Widgets")] on the startup ?

Reply:
That sounds good to me (and the same for Alias).

Reply:
@giannik we can't have one module depend on another module. We can have a module depend on a core project if another module. 

For this to work correctly, we would need to add a new Alias.Core project and move at AliasPart to it. Now you can depend on Alias.Core instead of the module

Reply:
is'nt it too much fuss just for a widget alternates . I think separating in a separate feature here : with [RequireFeatures("OrchardCore.Widgets")]  is  a nice balance.

Reply:
yeah sure. you can do that as well as long as it is only added when needed (i.e., using [RequireFeatures("OrchardCore.Widgets")])

Reply:
done